client/base/Router.js

```diff
@@ -1,21 +1,21 @@
-import React, { PropTypes } from 'react'
-import { Router as ReactRouter } from 'react-router'
-import { FixNamedRoutesSupport as allowNames } from 'react-router-named-routes'
+import React from 'react'
+import { Router as ReactRouter, useRouterHistory } from 'react-router'
+import { syncHistoryWithStore } from 'react-router-redux'
+import useNamedRoutes from 'use-named-routes'
+import createHistory from 'history/lib/createBrowserHistory'
 
-import { appRoutes } from 'modules/app'
+import { appRoutes as routes } from 'modules/app'
 
-allowNames(appRoutes)
+const history = useNamedRoutes(
```

Comment:
What if we moved the call to `useNamedRoutes` back to `client/index.js`?  That way, we could leave this code pretty much like it was and make more surgical changes over there.